### PR TITLE
FINERACT-1642: Bulk reversal of related Transactions

### PR DESCRIPTION
**Pull request by logoutdhaval**
## Description
[FINERACT-1642](https://issues.apache.org/jira/browse/FINERACT-1642).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Kindly include Swagger changes for the API

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountDomainServiceJpa.java [hunk 1]
```diff
@@ -292,15 +296,16 @@ private Map<BusinessEntity, Object> constructEntityMap(final BusinessEntity enti
...
 
         final boolean isSavingsInterestPostingAtCurrentPeriodEnd = this.configurationDomainService
                 .isSavingsInterestPostingAtCurrentPeriodEnd();
         final Integer financialYearBeginningMonth = this.configurationDomainService.retrieveFinancialYearBeginningMonth();
-
-        final Set<SavingsAccountChargePaidBy> chargePaidBySet = savingsAccountTransaction.getSavingsAccountChargesPaid();
-
+        Set<SavingsAccountChargePaidBy> chargePaidBySet = null;
+        for (SavingsAccountTransaction savingsAccountTransaction : savingsAccountTransactions) {
+            chargePaidBySet = savingsAccountTransaction.getSavingsAccountChargesPaid();
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountDomainServiceJpa.java]: This one looks  faulty. Overriding the set in each  iteration.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountDomainServiceJpa.java [hunk 2]
```diff
@@ -309,29 +314,31 @@ public SavingsAccountTransaction handleReversal(SavingsAccount account, SavingsA
...
         } else {
             account.calculateInterestUsing(mc, today, isInterestTransfer, isSavingsInterestPostingAtCurrentPeriodEnd,
                     financialYearBeginningMonth, postInterestOnDate, backdatedTxnsAllowedTill);
         }
 
-        List<SavingsAccountTransaction> newTransactions = new ArrayList<>();
-        newTransactions.add(reversal);
-
-        account.validateAccountBalanceDoesNotBecomeNegativeMinimal(savingsAccountTransaction.getAmount(), false);
+        account.validateAccountBalanceDoesNotBecomeNegativeMinimal(savingsAccountTransactions.get(0).getAmount(), false);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountDomainServiceJpa.java]: order is not guaranteed...

**Comment by logoutdhaval** [comment]
> the

**Comment by logoutdhaval** [comment]
> fynmanoj

done

**Review by fynmanoj — APPROVED** [review]
LGTM

**Pull request merged**